linera-service/src/client.rs

```diff
@@ -0,0 +1,1002 @@
+            network,
+        }
+    }
+
+    pub async fn project_new(&self, project_name: &str) -> Result<TempDir> {
+        let tmp = TempDir::new()?;
+        let mut command = self.run().await?;
+        assert!(command
+            .current_dir(tmp.path().canonicalize()?)
+            .kill_on_drop(true)
+            .arg("project")
+            .arg("new")
+            .arg(project_name)
+            .spawn()?
+            .wait()
+            .await?
+            .success());
+        Ok(tmp)
+    }
+
+    pub async fn project_publish<T: Serialize>(
+        &self,
+        path: PathBuf,
+        required_application_ids: Vec<String>,
+        publisher: impl Into<Option<ChainId>>,
+        argument: &T,
+    ) -> Result<String> {
+        let json_parameters = serde_json::to_string(&())?;
+        let json_argument = serde_json::to_string(argument)?;
+        let mut command = self.run_with_storage().await?;
+        command
+            .arg("project")
+            .arg("publish-and-create")
+            .arg(path)
+            .args(publisher.into().iter().map(ChainId::to_string))
+            .args(["--json-parameters", &json_parameters])
+            .args(["--json-argument", &json_argument]);
+        if !required_application_ids.is_empty() {
+            command.arg("--required-application-ids");
+            command.args(required_application_ids);
+        }
+        let stdout = Self::run_command(&mut command).await?;
+        Ok(stdout.trim().to_string())
+    }
+
+    pub async fn project_test(&self, path: &Path) {
+        let mut command = self.run().await.expect("TODO");
+        assert!(command
+            .current_dir(path)
+            .kill_on_drop(true)
+            .arg("project")
+            .arg("test")
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+            .unwrap()
+            .success());
+    }
+
+    async fn run(&self) -> Result<Command> {
+        let path = cargo_build_binary("linera").await;
+        let mut command = Command::new(path);
+        command
+            .current_dir(&self.tmp_dir.path().canonicalize()?)
+            .kill_on_drop(true)
+            .args(["--wallet", &self.wallet])
+            .args(["--send-timeout-us", "10000000"])
+            .args(["--recv-timeout-us", "10000000"])
+            .arg("--wait-for-outgoing-messages");
+        Ok(command)
+    }
+
+    async fn run_with_storage(&self) -> Result<Command> {
+        let mut command = self.run().await?;
+        command
+            .args(["--storage", &self.storage.to_string()])
+            .args([
+                "--max-pending-messages",
+                &self.max_pending_messages.to_string(),
+            ]);
+        Ok(command)
+    }
+
+    pub async fn create_genesis_config(&self) -> Result<()> {
+        assert!(self
+            .run()
+            .await?
+            .args(["create-genesis-config", "10"])
+            .args(["--initial-funding", "10"])
```

Comment:
At some point we should probably make these configurable, as optional arguments to `linera net up`.